addons/sale_mrp/tests/test_sale_mrp_flow.py

```diff
@@ -1539,3 +1539,111 @@ def test_product_type_service_1(self):
         mo = self.env['mrp.production'].search([('product_id', '=', finished_product.id)])
 
         self.assertTrue(mo, 'Manufacturing order created.')
+
+    def test_cancel_flow_1(self):
+        """ Sell a MTO/manufacture product.
+
+        Cancel the delivery and the production order. Then duplicate
+        the delivery. Another production order should be created."""
```

Comment:
if it's only to handle the duplicate usecase, i suggest to make a local patch in copy_data